### PR TITLE
Make Patcher tolerant of OpenAPIGetter being nil

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?
/kind regression

#### What this PR does / why we need it:
Make Patcher tolerant of OpenAPIGetter being nil.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix an issue where kubectl apply could panic when imported as a library
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by tomasaschan — APPROVED** [review]

**Comment by ardaguclu** [comment]
I assume that panicked cluster does not use OpenAPI v3 and fell back to v2 but I'm not sure how OpenAPIGetter can be nil. On the other hand, change looks good to me because we shouldn't panic any case;

/approve
/lgtm
/hold
in case you'd want another set of eyes, otherwise, feel free to unhold @Jefftree 

/triage accepted
/priority important-soon

**Pull request merged**
